lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -131,67 +126,105 @@ function dosomething_kudos_sort_by_taxonomy_term($a, $b) {
   return $result;
 }
 
-
 /**
- * Group all kudos in collection and organize them by taxonomy term.
+ * Group all kudos in collection and organize them by taxonomy term,
+ * including current users activity on retrieved kudos records.
  *
  * @param  array $data Array containing Kudos items.
- * @param  array $options
  * @return array
  */
-function dosomething_kudos_group_by_taxonomy_term($data, $options = []) {
-  $terms = [];
+function dosomething_kudos_group_by_taxonomy_term($data) {
+  $output = [];
+  $activity = [];
 
   foreach ($data as $item) {
     if (!($item instanceof Kudos)) {
       $item = (object) $item;
     }
 
-    if (!array_key_exists($item->term['id'], $terms)) {
-      $terms[$item->term['id']] = [
+    // If term is not in approved reactions list, skip it!
+    if (!in_array($item->term['name'], Kudos::getApprovedReactions())) {
+      continue;
+    }
+
+    // If the term is not already in the $output array then add it.
+    if (!array_key_exists($item->term['id'], $output)) {
+      $output[$item->term['id']] = [
         'term' => $item->term,
       ];
     }
 
-    $output = [
-      'id' => $item->id,
-      'user' => $item->user,
-    ];
-
-    if ($options['display'] === 'full') {
-      $output['reportback_item'] = $item->reportback_item;
+    if (!isset($output[$item->term['id']]['term']['total'])) {
+      $output[$item->term['id']]['term']['total'] = 0;
     }
 
-    if (isset($item->uri)) {
-      $output['uri'] = $item->uri;
-    }
+    $output[$item->term['id']]['term']['total'] += 1;
```

Comment:
See my _NOTE_ in **Any background context you want to provide?** I agree, but there's more shit to be done there and some caveats regarding then having still query against each item to see if the current user reacted, or grab the collection of current user reactions and compare those when each RB item is processed. There's more to consider and didn't want to break everything. This provides _a_ solution for right now, but ultimately I'd like to go down the SQL counting route.